circuits/cpp/barretenberg/cpp/src/barretenberg/stdlib/primitives/circuit_builders/circuit_builders.hpp

```diff
@@ -3,17 +3,24 @@
  * templates.
  */
 #pragma once
+#include "barretenberg/proof_system/circuit_builder/circuit_simulator.hpp"
 #include "barretenberg/proof_system/circuit_builder/standard_circuit_builder.hpp"
 #include "barretenberg/proof_system/circuit_builder/turbo_circuit_builder.hpp"
 #include "barretenberg/proof_system/circuit_builder/ultra_circuit_builder.hpp"
 
-template <typename T> concept HasPlookup = proof_system::IsAnyOf<T, proof_system::UltraCircuitBuilder>;
+template <typename T>
+concept HasPlookup =
+    proof_system::IsAnyOf<T, proof_system::UltraCircuitBuilder /* , proof_system::CircuitSimulatorBN254 */>;
+
+#define INSTANTIATE_STDLIB_SIMULATOR_METHOD(stdlib_method) template stdlib_method(proof_system::CircuitSimulatorBN254);
```

Comment:
As we discussed, these macros were just made to allow me to work incrementally. They should go away (perhaps that's already possible?).